drivers/infiniband/hw/ntrdma/ntrdma_vbell.c

```diff
@@ -211,8 +212,11 @@ static void ntrdma_dev_vbell_work(struct ntrdma_dev *dev, int vec)
 		ntrdma_vbell_head_fire(head, vbell_val);
 	}
 
-	if (ntc_clear_signal(dev->ntc, vec))
-		tasklet_schedule(&dev->vbell.work[vec]);
+	if (ntc_clear_signal(dev->ntc)) {
```

Comment:
Not sure what were the intentions here, each signal works from different affinity, if a change is needed here is to be more strict and make sure each irq is linked to it own vbells and not touch others